### PR TITLE
[JIT][CI] Integrate custom op tests with CI

### PR DESCRIPTION
**Pull request by goldsborough**
This PR is stacked on https://github.com/pytorch/pytorch/pull/10610, and only adds changes in one file `.jenkins/pytorch/test.sh`, where we now build the custom op tests and run them.

I'd also like to take this PR to discuss whether the [`TorchConfig.cmake`](https://github.com/pytorch/pytorch/blob/master/cmake/TorchConfig.cmake.in) I made is robust enough (we will also see in the CI) @orionr @Yangqing @dzhulgakov what do you think?

Also @ezyang for CI changes

### CODE REVIEW

**Review by orionr — APPROVED** [review]
Nicely done! Just one question, but looks great.

File: cmake/TorchConfig.cmake.in [hunk 1]
```diff
@@ -17,32 +17,44 @@
 #
 #   torch_add_custom_op_library(<name> <source_files>)
 
-SET(TORCH_ROOT "${CMAKE_CURRENT_LIST_DIR}/../")
+if ($ENV{TORCH_INSTALL_PREFIX})
+  set(TORCH_INSTALL_PREFIX ${TORCH_INSTALL_PREFIX})
```
[the remark this answers is not in the document]
Comment (goldsborough) [reply cmake/TorchConfig.cmake.in]: Yes, thanks!

**Review by zdevito — APPROVED** [review]
Stamp

**Pull request closed without merging**